### PR TITLE
docs/test: add UI test and long-form error docs for `E0461`

### PR DESCRIPTION
**Pull request by Ezrashaw**
Might take a couple of tries to pass CI. The UI test is x86-linux only; I'm not sure how to generalize it to other architectures. 

r? @GuillaumeGomez

### CODE REVIEW

File: compiler/rustc_error_codes/src/error_codes/E0461.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
+Couldn't find crate `..` with expected target triple `..`.
+
+Consider the following two files:
```
Comment (GuillaumeGomez) [inline compiler/rustc_error_codes/src/error_codes/E0461.md:3]: Please follow the format and use instead "Example of erroneous code".

**Comment by GuillaumeGomez** [comment]
Apart from the nit, seems good to me.

File: compiler/rustc_error_codes/src/error_codes/E0461.md (hunk 1 above)
Comment (Ezrashaw) [reply compiler/rustc_error_codes/src/error_codes/E0461.md]: Done. IMO, [RFC 1567](https://rust-lang.github.io/rfcs/1567-long-error-codes-explanation-normalization.html) is a bit outdated. It doesn't address these kinds of multi-file errors and other atypical errors. Also this really should be linted.

**Comment by Ezrashaw** [comment]
@GuillaumeGomez This completes another for #61137.
Additionally:
- `E0279` doesn't seem to be emitted anymore.
- `E0311` was completed in (somebody else) #100747
- `E0320`: (mine) #105792
- `E0377` was also completed by me (#105935) but it wasn't on the list.
- Finally, `E0457` and `E0460` got merged so they can be ticked.

Thanks for all your help, this has been a great first time contributing experience.

**Review by GuillaumeGomez — APPROVED** [review]

**Comment by GuillaumeGomez** [comment]
Thanks! Also thanks for the info. I'll update the issue accordingly.

@bors r+ rollup

**Comment by Ezrashaw** [comment]
@GuillaumeGomez Umm, not sure what has happened. My other PR made this unmergeable, and then when I fix the conflicts something happened with the cargo submodule.

**Comment by GuillaumeGomez** [comment]
Weird. Well, simplest way: `git reset --hard HEAD~1` (to remove your last commit) then re-make the changes (since they aren't many, should be fine).

**Comment by Ezrashaw** [comment]
Hmm, quite weird, considering my local git was completely happy. Should be good once CI passes.

**Review by GuillaumeGomez — APPROVED** [review]

**Comment by GuillaumeGomez** [comment]
Thanks!

@bors r+ rollup

**Comment by Ezrashaw** [comment]
@GuillaumeGomez I've removed the UI test on the basis that it is just too finicky to actually test the error code. I'm not even sure if the CI is able to test this error code. Sorry for the delay/ far too many CI attempts.

**Review by GuillaumeGomez — APPROVED** [review]

**Comment by GuillaumeGomez** [comment]
That's a shame but we can always give it another try later. Adding this explanation is already an improvement in any case. Thanks for working on this!

@bors r+ rollup

**Pull request merged**